### ISSUE TITLE
[BUG] A11y improvements for Dropdown

### ISSUE THREAD

**Issue by alxzak** [kirbydesign/designsystem#3683]
## Describe the bug

Please ensure that the dropdown component announces the item the focusfield is hovering over, when using keyboard. 
it only announces the item, when it is selected. 

Tech-refinement @loujuu

Following improvements should be considered to improve accessibility of the dropdown:

Considerations regarding best suited pattern for dropdown component:

When to Use the [Select-Only Combobox Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/) 
Use this pattern if your dropdown:
* Is triggered by a button or input
* Allows the user to select one item from a list
* Does not allow text input
* Behaves like a native <select> element, but with custom styling or templating. Users can type characters to select matching options.
* Uses a button to open a popover
* like an HTML <select>, users can type characters to select matching options.

When to Use the [Menu button](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/) Pattern
* A button that opens a menu 
* Offers a list of choices to the user, such as a set of actions or functions.
* Common use case: Context menus, toolbars, dropdowns with actions (e.g., “Edit”, “Delete”)

The Menu Button pattern is for triggering actions, not selecting values. It uses role="menu" and role="menuitem", which would mislead screen readers into thinking your dropdown is a command menu.

## Please complete the following information:
- OS: [e.g. Windows, MacOS, iOS]
- Browser [e.g. Chrome, Safari]
- Browser version [e.g. 22]

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/develop/.github/SUPPORT.md#team-kirby); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

**Comment by SimonCarlsen** [comment]
Det ser ud til at dropdown ikke implementerer listbox aria pattern korrekt!

Level Access scanning slår ud med Critical: "Element with role listbox, menu, or tree contains an element that is not allowed"

![Image](https://github.com/user-attachments/assets/ff495d73-8be6-48a3-b0f1-287971acba19)

**Comment by tmb-jyskefinans-dk** [comment]
> Det ser ud til at dropdown ikke implementerer listbox aria pattern korrekt!
> 
> Level Access scanning slår ud med Critical: "Element with role listbox, menu, or tree contains an element that is not allowed"
> 
> ![Image](https://github.com/user-attachments/assets/ff495d73-8be6-48a3-b0f1-287971acba19)

Er der noget attention på dette button elementet er ikke allowed i role="listbox"?

**Comment by SimonCarlsen** [comment]
De elementer der er tilgængelige i en dropdown annonceres ikke når man "navigere" imellem dem med pile-tasterne, derfor kan det være svært at vælge den rette option.

![Image](https://github.com/user-attachments/assets/c42c2d95-26ee-4d99-ba26-a06248ff95d8)

**Comment by Loujuu** [comment]
Decisions from tech refinement: @Fuzzy3 @RasmusKjeldgaard @Loujuu 
- Implement Select-Only-Combobox pattern
  - Add role="option" to default items
  - Add role="option" to custom template
- Unitest: Aria-attributes 
-  Add accessibility section to docs similar to menu https://cookbook.kirby.design/branch/kby-develop/#/home/showcase/menu
